### ISSUE TITLE
unauthorized request returns httpOK and mixed body

### ISSUE THREAD

**Issue by zyclonite** [BTBurke/caddy-extauth#1]
Caddyfile
```
host.domain.tld {
  extauth http://127.0.0.1:8081/api/check
  proxy / http://127.0.0.1:8080/protected/ {
    transparent
  }
}
```

having a proxy resource returning a non 200 code (/api/check)

original
```
code: 401
body: {"message":"Unauthorized"}
```

on the protected resource (after extauth processing)
```
code: 200
body: {"message":"Unauthorized"}401 Unauthorized
```

**Comment by BTBurke** [comment]
Thanks, I hadn't realized this plugin had gone live while I was still testing it.  Can you post what you're doing in your /api/check handler so I can see how the headers are being set?

**Comment by BTBurke** [comment]
I just pushed what I think is a fix in v0.6.0 which should signal to downstream handlers that the header 401 status is already written and not to write any further to the request.  Can you check it out and see if that fixes the issue?  Thanks for testing this.  I'm still working on it a little bit.

**Comment by zyclonite** [comment]
now it works, thank you
